Code/6-wire-version/src/main.cpp

```diff
@@ -167,7 +244,114 @@ void sendMQTT(){
 
 
 
-void startOTA() { // Start the OTA service
+/**
+ * Start a Wi-Fi access point, and try to connect to some given access points.
+ * Then wait for either an AP or STA connection
+ */
+void startWiFi()
+{
+  //WiFi.mode(WIFI_STA);
+  WiFi.setAutoReconnect(true);
+  WiFi.persistent(true);
+  WiFi.hostname(netHostname);
+
+  if (enableStaticIp4)
+  {
+    Serial.println("WiFi > using static IP \"" + ip4Address.toString() + "\" on gateway \"" + ip4Gateway.toString() + "\"");
+    WiFi.config(ip4Address, ip4Gateway, ip4Subnet, ip4DnsPrimary, ip4DnsSecondary);
+  }
+
+  if (enableAp)
+  {
+    Serial.println("WiFi > using WiFi configuration with SSID \"" + apSsid + "\"");
+
+    WiFi.begin(apSsid, apPwd);
+
+    Serial.print("WiFi > Trying to connect ...");
+    int maxTries = 10;
+    int tryCount = 0;
+
+    while (WiFi.status() != WL_CONNECTED)
+    {
+      delay(1000);
+      Serial.print(".");
+      tryCount++;
+
+      if (tryCount >= maxTries)
+      {
+        Serial.println("");
+        Serial.println("WiFi > NOT connected!");
+        if (enableWmApFallback)
+        {
+          // disable specific WiFi config
+          enableAp = false;
+          enableStaticIp4 = false;
+          // fallback to WiFi config portal
+          startWiFiConfigPortal();
+        }
+        break;
+      }
+      Serial.println("");
+    }
+  }
+  else
```

Comment:
this will only run once - right when the system goes up the first time (or on the case when aborting the fallback feature and replug to let it boot again..).